File: p2.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n=len(citations)
        bucket=[0]*(n+1)
        #bucket sort
        for i in range(n):
            if(citations[i]>=n):
                bucket[n]+=1
            else:
                bucket[citations[i]]+=1
        summ=0
        for i in range(n,-1,-1):
            summ+=bucket[i]
            if (summ>=i):
                return i

File: p2.java
// Time Complexity : O(n + nlog(n))
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no


// Your code here along with comments explaining your approach

class Solution {
    public int hIndex(int[] citations) {
        //Sort the array
        Arrays.sort(citations);
        //Get to the point where number of papers on right is equal to or less than citaions
        for(int i=citations.length; i>0; i--){
            if(i <= citations[citations.length-i]) return i;
        }
        return 0;
    }
}